socorro/external/es/crashstorage.py

```diff
@@ -53,6 +57,61 @@ def is_valid_key(key):
     return bool(VALID_KEY.match(key))
 
 
+def remove_bad_keys(data):
+    """Removes keys from the top-level of the dict that are bad
+
+    Good keys satisfy the following properties:
+
+    * have one or more characters
+    * are composed of a-zA-Z0-9_-
+
+    Anything else is a bad key and needs to be removed.
+
+    Note: This modifies the data dict in-place and only looks at the top level.
+
+    :arg dict data: the data to remove bad keys from
+
+    """
+    if not data:
+        return
+
+    # Copy the list of things we're iterating over because we're mutating
+    # the dict in place.
+    for key in list(data.keys()):
+        if not is_valid_key(key):
+            del data[key]
```

Comment:
This function didn't need to be part of the crash storage class, so I moved it to make it easier to test in the future if we ever change it.